rusha.sweet.js

```diff
@@ -509,16 +509,51 @@
   // messages containing a jobid and a buffer
   // or blob object, and return the hash result.
   if (typeof FileReaderSync !== 'undefined') {
-    var reader = new FileReaderSync(),
-        hasher = new Rusha(4 * 1024 * 1024);
-    self.onmessage = function onMessage (event) {
-      var hash, data = event.data.data;
+    var hasher;
+    var readTotal = 0;
+    var BLOCK_SIZE;
+    var hashData = function hash (data, cb) {
       try {
-        hash = hasher.digest(data);
-        self.postMessage({id: event.data.id, hash: hash});
+        return cb(null, hasher.digest(data));
       } catch (e) {
-        self.postMessage({id: event.data.id, error: e.name});
+        return cb(e);
+      }
+    };
+    var hashFile = function hashArrayBuffer (file, cb) {
+      var reader = new self.FileReader()
+      reader.onloadend = function onloadend () {
+        var buffer = reader.result
+        readTotal += reader.result.byteLength
+        try {
+          hasher.append(buffer);
+        }
+        catch (e) {
+          cb(e);
+          return;
+        }
+        if (readTotal < file.size) {
+          hashFile(file, cb);
+        } else {
+          cb(null, hasher.end());
+        }
       }
+      reader.readAsArrayBuffer(file.slice(readTotal, readTotal + BLOCK_SIZE))
+    };
+
+    self.onmessage = function onMessage (event) {
+      BLOCK_SIZE = event.data.BLOCK_SIZE || (4 * 1024 * 1024);
```

Comment:
The property name here should probably be lowercase, since it is not a constant at all. Also, there doesn't seem to be a reason here to make the variable scope larger than necessary.